contrib/functions/go/annotate-apply-time-mutations/pkg/types.go

```diff
@@ -0,0 +1,25 @@
+package pkg
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cli-utils/pkg/object/mutation"
+)
+
+// ApplyTimeMutation is a Kubernetes resource that allows specifying mutations
+// using a seperate KRM object, instead of an annotation string on the target
+// object.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ApplyTimeMutation struct {
+	// TODO: Use v1.TypeMetaApplyConfiguration from client-go instead?
```

Comment:
Are there any difference between them?

Reply:
Looks like the -ApplyConfiguration resources in client-go are purely for server-side apply, making every field a pointer. Since ApplyTimeMutation never gets sent to the server, we don't need server-side apply. So I can remove the TODO.